.github/workflows/test-and-publish-nightly.yml

```diff
@@ -28,44 +28,73 @@ jobs:
   build-binaries-and-test:
     strategy:
       matrix:
-        runson:
-          - display: linux-x64
-            name: ubuntu-latest # Using "latest" here as the build and test will any ways run inside a container which we control
-        java-version: [11, 17, 21, 24]
-        java-distribution: [corretto]
-        container: ["public.ecr.aws/async-profiler/asprof-builder-x86:latest"]
```

Comment:
Is there a way to keep this matrix simplification and not repeat these in each configuration below?

Reply:
Unfortunately the matrix is a cartesian product, thus we cannot put `published` here. I see only a few solutions:
- YAML anchors (not supported in GitHub Actions yet)
- Flattening the matrix as I did in the most recent revision
- Having a set of filtering variables like `RELEASE_FROM_JAVA_VERSION`
- Preparing a JSON (can be static, or dynamic as the output of another job) containing a list of configurations (OS, JDK, JDK version, publish? ), something similar to what they do [here](https://stackoverflow.com/a/65434401)

Let me know which one you prefer @krk @apangin

Reply:
Do we ever want to publish from more than 1 configuration?

Reply:
Also, we could add more versions and distributions other than corretto for testing in the future - sounds like we actually need a cartesian product here, rather than duplicating entries.

Reply:
Sure, last revision restored the matrix